demystifying_machine_learning/demystifying_machine_learning.md

```diff
@@ -0,0 +1,712 @@
+
+We've talked at length about bias in terms of **mathematical bias** in the [bias-variance tradeoff](#the-bias-variance-tradeoff), but there's another kind of bias that you need to consider in machine learning:
+
+Many of the large datasets that might be analyzed with machine learning reflect the unfair and immoral inequalities inherent in our society; models trained on data with that kind of bias will reproduce that bias in their predictions and suggestions.
+That means that it's very easy to unintentionally create machine learning systems that enforce racist/sexist/ableist reasoning, and that may have terrible consequences if applied in the real world.
+
+<div class = "learn-more">
+<b style="color: rgb(var(--color-highlight));">Learning connection</b><br>
+
+To learn more about one particularly salient example of this problem, read ["Dissecting racial bias in an algorithm used to manage the health of populations"](https://www.science.org/doi/full/10.1126/science.aax2342) by Ziad Obermeyer and colleagues. From the abstract:
+
+> The U.S. health care system uses commercial algorithms to guide health decisions. Obermeyer et al. find evidence of racial bias in one widely used algorithm, such that Black patients assigned the same level of risk by the algorithm are sicker than White patients (see the Perspective by Benjamin). The authors estimated that this racial bias reduces the number of Black patients identified for extra care by more than half. Bias occurs because the algorithm uses health costs as a proxy for health needs. Less money is spent on Black patients who have the same level of need, and the algorithm thus falsely concludes that Black patients are healthier than equally sick White patients. Reformulating the algorithm so that it no longer uses costs as a proxy for needs eliminates the racial bias in predicting who needs extra care.
+
+You can also listen to the first author, [Ziad Obermeyer, talking about this paper at a workshop on algorithmic fairness](https://www.youtube.com/watch?v=u34e9CYIMZs).
+
+</div>
+
+Mathematician Cathy O'Neil has done a lot to bring public attention to issues of bias and unfairness in machine learning algorithms used in the sectors like education, finance, and the justice system.
+For an excellent overview of the problem, watch her TED talk:
+
+!?[Cathy O'Neil's TED Talk, "The era of blind faith in big data must end".](https://www.youtube.com/watch?v=_2u_eHHzRto)
+
+**What should you do about unintended biases in your own algorithms?**
+
+As both O'Neil and Obermeyer emphasize, algorithms can be interrogated.
+In addition to assessing how accurate a model is, you can and should look for biases in your results.
+
+For example, if you expect that your model's predictions shouldn't be systematically different for patients on [Medicaid](https://www.hhs.gov/answers/medicare-and-medicaid/who-is-eligible-for-medicaid/index.html) vs. private insurance, then check the output to see if that's the case.
+
```

Comment:
```suggestion
## Model explainability

In machine learning, especially on big data, sometimes models are so complex in the way they combine and perform math on features that it's not clear in human terms why a particular prediction was made.  This feels very different from simple statistical inference, which can usually be described in a sentence or two: "as number of school days lost to illness rises, patient-reported quality of life scores fall in a linearly inverse relationship."  

Particularly complex models which are hard to explain because they use many steps or do advanced mathematical techniques are called "black boxes".  They can be very accurate for prediction and can do a good job of finding relationships that humans would miss, as they combine hundreds of data points on millions of observations.  An example of a black box algorithm might be credit card anomaly detection.  If you've ever had someone say to you, "well, it's not clear why we thought your credit card was stolen, but there was just something about your purchases and travel that made us automatically block your card," you have experienced the unexplainability of a black box algorithm.

Explainability is less important in some fields and more important in others.  If you are creating models that require human buy-in or are aimed at helping people change behaviors to reduce risk, explainability is paramount.  

For example, consider a black box model of cancer risk, which takes hundreds of variables including where people lived at different points in their life, jobs they've had, lab values, marital status, diet and exercise habits, genomic data, known heavy metal exposures, etc.  This is a staggeringly complex model that can accurately predict solid tumor risk, but because it combines and recombines data and uses strategies that are hard to describe to people without computer science PhDs, it's not explainable.  That means that if a patient asks, "will reducing my red meat consumption improve my cancer risk?" there's no way to know how that particular variable affects the final model outcome.

Black box models can also be unfair in ways that are obscured, and result in giving results that are biased in terms of race, gender, or other category of human bias.  Demonstrating and ameliorating bias in black box models can be tricky because of the number of potentially biased features and the complex algorithm that processes those features.

Explainable machine learning (also known as explainable AI) can be achieved by:

 * Using algorithms that are simpler and more intuitive (for example, using decision trees instead of neural networks)
 * Simulating the results of changing a single feature to measure how much that feature influences the prediction, and how
 * Using software or code libraries that can show the relative salience or weight of various features


```

I feel like at least talking about explainability is important but please veto some/any/all of this text.  There's a decent video at https://www.youtube.com/watch?v=Yg3q5x7yDeM that I think threads the needle of brevity but complexity, too...

Reply:
I think this is a great addition! I think there are just a couple little things I might tweak, but I'll put it in almost as-written. Thank you so much for drafting! I think this will make it a much more valuable module